pkg/controller/migmigration/migrate.go

```diff
@@ -93,42 +80,52 @@ func (r *ReconcileMigMigration) migrate(migration *migapi.MigMigration) (int, er
 		return 0, err
 	}
 
-	migration.Status.SetCondition(migapi.Condition{
-		Type:     Running,
-		Status:   True,
-		Reason:   task.Phase.Name,
-		Category: Advisory,
-		Message:  RunningMessage,
-	})
-
 	// Result
-	migration.Status.Phase = task.Phase.Name
-	if task.Phase.Equals(WaitOnResticRestart) {
-		return 10, nil
-	}
-	if task.Phase.Final() {
+	migration.Status.Phase = task.Phase
+
+	// Succeeded
+	if task.Phase == Completed {
 		migration.Status.DeleteCondition(Running)
-		migration.Status.SetCondition(migapi.Condition{
-			Type:     Succeeded,
-			Status:   True,
-			Reason:   task.Phase.Name,
-			Category: Advisory,
-			Message:  SucceededMessage,
-		})
+		failed := task.Owner.Status.FindCondition(Failed)
+		if failed == nil {
```

Comment:
phase refactor merge

Reply:
Upon completion, either the existing _Failed_ condition Status is set True, or the _Succeeded_ condition is set.